feat(plugins/i18next): Use path to decide default namespace

MrTwixxy opened the pull request:
_Disclaimer: I haven't tested it yet, since Im unable to build myself_

This fixes the default behavior of using the first namespace when there's none defined.
Useful when you have a monorepo and only want 1 sherlock project

felixhaeberle commented:
@MrTwixxy can you sign the CLA?

MrTwixxy commented:
I have read the CLA Document and I hereby sign the CLA

felixhaeberle commented:
@MrTwixxy I've approved the pipeline so we can check if the changes leads to errors.

MrTwixxy commented:
I've fixed the tests and added a new one. 
The behavior isn't 'defined' somewhere, but the default behavior isn't either. 
I don't expect things to break for the following reasons:
 - Nothing changed if namespaces are used how they normally are
 - If you aren't using them how they normally are, it's just using always using the default, which still happens if the path doesnt match any namespace

felixhaeberle commented:
hey, we have to discuss this change internally as this would mean that every plugin now needs some kind of strategy on how to do `documentPath`'s, see https://github.com/MrTwixxy/inlang-monorepo-fork/blob/7ec9773714d9875bab0b3778a7567b4b103f1aed/inlang/source-code/versioned-interfaces/plugin/src/customApis/app.inlang.ideExtension.ts#L43

MrTwixxy commented:
Having the argument passed into the function doesn't necessarily mean that it has to be used right?

samuelstroschein commented:
@felixhaeberle i tend to agree with @MrTwixxy. just because the argument exists, doesn't mean that a plugin needs to use `documentPath`.

samuelstroschein commented:
i propose to close this pr given that we will re-implement the custom API https://github.com/opral/inlang-sdk/issues/125
